src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -21,25 +32,471 @@ const ActorInit En_Bombf_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/EnBombf_Init.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x29, 0x39, 0x20, COLSHAPE_CYLINDER },
+    { 0x02, { 0x00000000, 0x00, 0x00 }, { 0x0003F828, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 9, 18, 10, { 0, 0, 0 } },
+};
+
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
+    {
+        { 0x00, { 0x00000008, 0x00, 0x08 }, { 0x00000000, 0x00, 0x00 }, 0x19, 0x00, 0x00 },
+        { 0, { { 0, 0, 0 }, 0 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x39, 0x00, 0x00, 0x00, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+extern Gfx D_06000340[];
+extern Gfx D_06000408[];
+extern Gfx D_06000530[];
+
+void EnBombf_SetupAction(EnBombf* this, EnBombfActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnBombf_Init(Actor* thisx, GlobalContext* globalCtx) {
+    f32 shapeUnk10 = 0.0f;
+    s32 pad;
+    EnBombf* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    this->unk_200 = 1;
+    Collider_InitCylinder(globalCtx, &this->bombCollider);
+    Collider_InitJntSph(globalCtx, &this->jntSphList);
+    Collider_SetCylinder(globalCtx, &this->bombCollider, &this->actor, &sCylinderInit);
+    Collider_SetJntSph(globalCtx, &this->jntSphList, &this->actor, &sJntSphInit, &this->explosionCollider[0]);
+
+    if (this->actor.params == 0) {
+        shapeUnk10 = 1000.0f;
+    }
+
+    ActorShape_Init(&this->actor.shape, shapeUnk10, ActorShadow_DrawFunc_Circle, 12.0f);
+    this->actor.posRot2.pos = this->actor.posRot.pos;
+
+    if (Actor_FindNearby(globalCtx, &this->actor, ACTOR_BG_DDAN_KD, ACTORTYPE_BG, 10000.0f) != 0) {
+        this->actor.flags |= 0x20;
+    }
+
+    this->actor.colChkInfo.unk_10 = 10.0f;
+    this->actor.colChkInfo.unk_12 = 10;
+    this->actor.unk_1F = 0;
+
+    if (this->actor.params == 0) {
+        this->timer = 140;
+        this->flashSpeedScale = 15;
+        this->actor.gravity = -1.5f;
+        Actor_ChangeType(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORTYPE_EXPLOSIVES);
+        this->actor.colChkInfo.mass = 200;
+        this->actor.flags &= ~1;
+        EnBombf_SetupAction(this, EnBombf_Move);
+    } else {
+        this->actor.colChkInfo.mass = 0xFF;
+        this->bumpOn = true;
+        this->flowerBombScale = 1.0f;
+        EnBombf_SetupGrowBomb(this, thisx->params);
+    }
+
+    this->actor.uncullZoneScale += 31000.0f;
+    this->actor.uncullZoneForward += 31000.0f;
+}
+
+void EnBombf_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBombf* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->bombCollider);
+    Collider_DestroyJntSph(globalCtx, &this->jntSphList);
+}
+
+void EnBombf_SetupGrowBomb(EnBombf* this, s16 params) {
+    EnBombf_SetupAction(this, EnBombf_GrowBomb);
+}
+
+void EnBombf_GrowBomb(EnBombf* this, GlobalContext* globalCtx) {
+    EnBombf* bomb;
```

Comment:
```suggestion
    EnBombf* bombFlower;
```

Reply:
I was making it clear that its an actual bomb and not the flower. but i guess thats obvious in context